specs/PushNotifications/PushNotifications-spec.md

```diff
@@ -2,120 +2,150 @@
+-   Supporting Visual Cloud Toasts for packaged Win32 apps.
+-   Supporting Visual Cloud Toasts for unpackaged Win32 apps.
+
+A Portal Registration flow through
+[AAD (Azure Active Directory)](https://docs.microsoft.com/azure/active-directory/develop/quickstart-register-app)
+will also be defined that removes the dependency of the Push Flow with <br/> the Partner Center
+Portal. The RemoteIdentifier GUID in this spec maps to the ApplicationId(ClientId) in the AAD App
+<br/> Registration process. Below is an illustration of the proposed flow through AAD:<br/>
 ![Push Flow For Windows App SDK](Push.png)
 
-Phase 1 (Windows App SDK Preview):<br/>
-WNS Push APIs will only support WIN32 Packaged app (MSIX Desktop Bridge).
+Phase 1 (Windows App SDK Preview):<br/> WNS Push APIs will only support WIN32 Packaged app (MSIX
+Desktop Bridge).
 
-Phase 2 (Windows App SDK V1):<br/>
-WNS Push APIs will support unpackaged Win32 scenarios.
+Phase 2 (Windows App SDK V1):<br/> WNS Push APIs will support unpackaged Win32 scenarios.
 
-Link to the official Windows App SDK timeline can be found [here](https://github.com/microsoft/WindowsAppSDK/blob/main/docs/roadmap.md).
+Link to the official Windows App SDK timeline can be found
+[here](https://github.com/microsoft/WindowsAppSDK/blob/main/docs/roadmap.md).
 
 # Examples
 
 ## In this scenario, the process that Registers the Push Activator and the process specified as the COM server are the same
+
 The code in Main would follow the pattern below:
-* The Registration will take in a CLSID of the COM server as part of the Activator setup. The registration API will simply be <br/>
-a thin wrapper around the BackgroundInfra WinRT APIs and abstract away the COM details from the developer including registration <br/>
-of the inproc Windows App SDK component.
-* The app will query ApplifeCycle APIs to retrieve an ActivationKind. (Note: This is covered in a seperate AppLifeCycle API spec.)
-* If the Activation is Push, the app will QI the Arguments property to retrieve an instance of PushActivatedEventArgs and get <br/>
-the Push payload from it. Care needs to be taken to Get a Deferral and Complete the Deferral while processing the payload for <br/>
-Power Management compliance.
-* If the Activation is Foreground, the app will do 2 things:
-  * It will request a Push Channel Asynchronously with an implementation of Progress and Completed event handler.
-    * Expect Channel operations to take around 2-16 minutes in some rare cases (retryable errors). In the 90th percentile case, <br/>
-    it should be fairly quick operation (in a few seconds).
-  * It will subscribe to a In-memory Push event handler hanging off the retrieved Push Channel component.
-    * The app should ideally handle the event by setting the Handled property to true to prevent Background Activation from <br/>
-    launching a new process.
+
+-   The Registration will take in a CLSID of the COM server as part of the Activator setup. The
```

Comment:
(No change required, note on writing style)

Instead of "the bar will foo" consider using "the bar foos".  "Will" suggests indeterminate future, while "bar foos" is a stronger statement that it must be so.  This document, once merged, should be read as "now" rather than "some future point." So someone who comes across this in the future doesn't need to wonder "have they already made this happen, can I use it now?" and instead says "ok, I need my bar to foo in order to accomplish this."